unit_test/solver/network/utils/test_fluid_properties.py

```diff
@@ -15,196 +15,183 @@
-        IE = 0
+        x = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+        order = 5
+        y = [value**order for value in x]
+        interpolation = Interpolation(x, y, order)
 
         # Act
-        T = self.fluid_props.get_t(IE)
+        interpolated_value = interpolation(2)
 
         # Assert
-        self.assertEqual(T, 1)
+        self.assertAlmostEquals(interpolated_value, 32)
 
-    def test_get_temperature_with_invalid_internal_energy(self):
-        """Test the get_temperature method with an invalid internal energy."""
+    def test_validation_error(self):
+        """Test the validation error method of the Interpolation class."""
         # Arrange
-        IE = -1
+        x = [1, 2, 3]
+        y = [2, 4, 6]
+        order = 1
+        interpolation = Interpolation(x, y, order)
+        interpolation.coefficients = np.array([0, 1])
 
         # Act
         with self.assertRaises(ValueError) as cm:
-            self.fluid_props.get_t(IE)
+            interpolation._check_interpolation()
 
         # Assert
-        self.assertEqual(
-            str(cm.exception),
-            "The internal energy -1 is outside the range of the fluid properties.",
-        )
+        self.assertEqual(str(cm.exception), "Interpolation error: error is more then 2%.")
         self.assertIsInstance(cm.exception, ValueError)
 
-    def test_get_density(self):
-        """Test the get_density method."""
+    def test_check_bounds(self):
+        """Test the check bounds method of the Interpolation class."""
         # Arrange
-        T = 1
+        x = [1, 2, 3]
+        y = [2, 4, 6]
+        order = 1
+        interpolation = Interpolation(x, y, order)
 
         # Act
-        rho = self.fluid_props.get_density(T)
+        with self.assertRaises(ValueError) as cm:
+            interpolation._check_bounds(0)
 
         # Assert
-        self.assertEqual(rho, 3)
+        self.assertEqual(str(cm.exception), "Value is out of bounds.")
+        self.assertIsInstance(cm.exception, ValueError)
 
-    def test_get_density_with_invalid_temperature(self):
-        """Test the get_density method with an invalid temperature."""
-        # Arrange
-        T = 0
 
-        # Act
-        with self.assertRaises(ValueError) as cm:
-            self.fluid_props.get_density(T)
+class OmotesFluidPropertiesTest(unittest.TestCase):
+    """Test the OmotesFluidProperties class."""
 
-        # Assert
-        self.assertEqual(
-            str(cm.exception), "The temperature 0 is outside the range of the fluid properties."
-        )
-        self.assertIsInstance(cm.exception, ValueError)
+    def setUp(self):
+        """Set up the test."""
+        self.omotes_fluid_properties = OmotesFluidProperties()
 
-    def test_get_viscosity(self):
-        """Test the get_viscosity method."""
+    def test_init(self):
+        """Test the initialization of the OmotesFluidProperties class."""
         # Arrange
-        T = 1
+        T = [i + 273.15 for i in range(150)]
```

Comment:
You don't set this range of temp values; should that be an input property of the class?

Reply:
Added this as input